pkg/forklift-api/webhooks/validating-webhook/admitters/secret-admitter.go

```diff
@@ -58,6 +79,30 @@ func (admitter *SecretAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissio
 		return webhookutils.ToAdmissionResponseAllow()
 	}
 
+	if resourceType, ok := admitter.secret.GetLabels()["createdForResourceType"]; ok {
+		switch resourceType {
+		case "hosts":
+			if hostName, ok := admitter.secret.GetLabels()["createdForResource"]; ok {
+				if tested, err := admitter.validateHostSecret(hostName); err != nil {
+					if tested {
+						return &admissionv1.AdmissionResponse{
```

Comment:
Just a suggestion, how about editing the `ToAdmissionResponseError` function to have the option to send custom code, and if empty use default `http.StatusBadRequest`.
Or alternatively adding a new function with the code as a parameter.

Reply:
yeah, that can be done separately